src/test/scala/mesosphere/mesos/TaskBuilderTest.scala

```diff
@@ -619,24 +621,55 @@ class TaskBuilderTest extends MarathonSpec {
   test("PortsEnvWithOnlyPorts") {
     val command =
       TaskBuilder.commandInfo(
+        app = AppDefinition(
+          ports = Seq(8080, 8081)
+        ),
+        taskId = Some(TaskID("task-123")),
+        host = Some("host.mega.corp"),
+        ports = Seq(1000, 1001),
+        envPrefix = None
+      )
+    val env: Map[String, String] =
+      command.getEnvironment.getVariablesList.asScala.toList.map(v => v.getName -> v.getValue).toMap
+
+    assert("1000" == env("PORT_8080"))
+    assert("1001" == env("PORT_8081"))
+  }
+
+  test("PortsEnvWithCustomPrefix") {
+    val command =
+      TaskBuilder.commandInfo(
         AppDefinition(
           ports = Seq(8080, 8081)
         ),
         Some(TaskID("task-123")),
         Some("host.mega.corp"),
-        Seq(1000, 1001)
+        Seq(1000, 1001),
+        Some("CUSTOM_PREFIX_")
       )
     val env: Map[String, String] =
       command.getEnvironment.getVariablesList.asScala.toList.map(v => v.getName -> v.getValue).toMap
 
-    assert("1000" == env("PORT_8080"))
-    assert("1001" == env("PORT_8081"))
+    assert("1000,1001" == env("CUSTOM_PREFIX_PORTS"))
+
+    assert("1000" == env("CUSTOM_PREFIX_PORT"))
+
+    assert("1000" == env("CUSTOM_PREFIX_PORT0"))
+    assert("1000" == env("CUSTOM_PREFIX_PORT_8080"))
+
+    assert("1001" == env("CUSTOM_PREFIX_PORT1"))
+    assert("1001" == env("CUSTOM_PREFIX_PORT_8081"))
+
+    assert("host.mega.corp" == env("CUSTOM_PREFIX_HOST"))
+
+    assert(Seq("HOST", "PORTS", "PORT0", "PORT1").forall(k => !env.contains(k)))
+    assert(Seq("MESOS_TASK_ID", "MARATHON_APP_ID", "MARATHON_APP_VERSION").forall(env.contains))
```

Comment:
I would rather check for all variables that do not start with the prefix and ascertain that all exceptions are in the list.

If you add an unprefixed variable now, no test will fail.